### PR TITLE
feat(InputPassword): Add InputPassword component

### PR DESCRIPTION
**Pull request by sizvix**
Add InputPassword witch purpose a button to show/hide the string in the input.
The button could be disabled with the props "askShowHide" to "false".

### CODE REVIEW

**Comment by Crash--** [comment]
Can you add : 
- snapshot 
- an exemple

**Comment by GoOz** [comment]
There's the how to in the PR template 

```
yarn build:doc:react
yarn deploy:doc --repo git@github.com:USERNAME/cozy-ui.git
```

**Comment by sizvix** [comment]
@GoOz : That was one of our reflexions, but it need to transform Field witch is a simple function to a class extending React component , just for one face of Field ... ( but that's simple ... )
Or just wrap the field in an element witch extends React component.
The way I prefer could be have a place after the label for generic "inputInteractionElement" , witch here is "show/hide buton" , on a `type="date"` it could be "today" , etc ...

File: react/InputPassword/styles.styl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.wrapInputPassword
```
Comment (gregorylegarec) [inline react/InputPassword/styles.styl]: We use to use dash-case for or CSS class names. We are also prefixing or component related class names with `c-`.

Actually it was not in our guidelines, I juste made a PR to fix that. https://github.com/cozy/cozy-guidelines/pull/57

File: react/InputPassword/index.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import cx from 'classnames'
+import { default as LabelStyles } from '../Label/styles.styl'
+import Input from '../Input'
+import styles from './styles.styl'
+
+class InputPassword extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { visible: false }
+    this.askShowHide = props.askShowHide !== false
```
Comment (gregorylegarec) [inline react/InputPassword/index.jsx]: I think `visibilityButton` (`true`/`false`) is a better name for this prop. What do you think ?

By the way it should be only taken into account in the render method.

File: react/InputPassword/index.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  }
+  render() {
+    return (
+      <div className={styles['wrapInputPassword']}>
+        {this.askShowHide && (
+          <div
+            className={cx(LabelStyles['c-label'], styles['actionOnInput'])}
+            onClick={() => this.toggleVisibility()}
+          >
+            {this.state.visible ? 'HIDE' : 'SHOW'}
```
Comment (gregorylegarec) [inline react/InputPassword/index.jsx]: We cannot let untranslated text anywhere in our code.

Also, this line ask a question : we need to translate component from Cozy-UI and it has never been done before. A quick solution could be to pass the label as props (maybe `buttonHideLabel` and `buttonShowLabel`), waiting for a more global solution provided by Cozy-UI.

File: react/InputPassword/index.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          </div>
+        )}
+        <Input
+          {...this.props}
+          type={this.state.visible ? 'text' : 'password'}
```
Comment (gregorylegarec) [inline react/InputPassword/index.jsx]: As `this.state.visible` is used two times in this methond, maybe we can retrieve it at the beginning of the method with 
```js
const { visible } = this.state
```

File: react/InputPassword/styles.styl (hunk 1 above)
Comment (GoOz) [reply react/InputPassword/styles.styl]: https://docs.cozy.io/cozy-ui/styleguide/ there is guidelines but not in cozy-guidelines I admit.

File: react/InputPassword/index.jsx (hunk 3 above)
Comment (ptbrowne) [reply react/InputPassword/index.jsx]: Finally the `translate` signature could serve to something :) Default translating strings could be passed directly as options

```
export default translate({
  en: { show: 'Show', hide: 'Hide' },
  fr: { show: 'Montrer', hide: 'Cacher' }, 
})(InputPassword)
```

[the remark this answers is not in the document]
Comment (sizvix) [reply react/InputPassword/index.jsx]: temporary with props ...

File: react/InputPassword/styles.styl (hunk 1 above)
Comment (sizvix) [reply react/InputPassword/styles.styl]: @GoOz is it good now ?

Comment (GoOz) [reply react/InputPassword/styles.styl]: where is the link?

Comment (sizvix) [reply react/InputPassword/styles.styl]: [new styles.styl](https://github.com/cozy/cozy-ui/blob/fc7206c7b78148dffc48ad519e77f8a042717513/react/Field/styles.styl)
I add the classes : `.wrap-input-password` and `.action-on-input`  :)

File: react/Field/index.jsx [hunk 5]
```diff
@@ -67,25 +124,29 @@ const Field = props => {
...
   value: PropTypes.string,
   placeholder: PropTypes.string,
   error: PropTypes.bool,
-  size: PropTypes.oneOf(['tiny', 'medium', 'large'])
+  size: PropTypes.oneOf(['tiny', 'medium', 'large']),
+  secondaryLabels: PropTypes.object
 }
 
 Field.defaultProps = {
+  enableExtraFunction: true,
```
Comment (gregorylegarec) [inline react/Field/index.jsx]: This name is not very explicit and can be confusing as we can expect a function here.

After discussing with @ptbrowne, the best way to achieve this seems to pass `showVisibilityButton` inside a `fieldProps` prop.

**Comment by gregorylegarec** [comment]
As there were only a few remark remaining I made the change directly. Does @GoOz and @Crash--  may approve changes ? thanks !

File: react/Field/index.jsx [hunk 6]
```diff
@@ -34,10 +79,23 @@ const Field = props => {
...
+          <InputPassword
+            id={id}
+            type={type}
+            placeholder={placeholder}
+            value={value}
+            error={error}
+            onChange={onChange}
+            readOnly={readOnly}
+            {...fieldProps}
+            {...secondaryLabels}
```
[the remark this answers is not in the document]
Comment (gregorylegarec) [reply react/Field/index.jsx]: I think that it shows a bad pattern at some place. Maybe InputPassword should be passed as prop to Field. It implies a little rework of our Field component.

**Comment by gregorylegarec** [comment]
Thanks @sizvix 👍

**Pull request merged**